docs/dump_heapdump.md

```diff
@@ -24,25 +24,354 @@
+Heap dumps are generated by default in PHD format when the Java heap runs out of space. If you want to trigger the production of a heap dump in response to other situations, or in classic format, you can use one of the following options:
+
+- Configure the **heap** dump agent. For more information, see the [-Xdump](xdump.md) option.
+- Use the `com.ibm.jvm.Dump` API programmatically in your application code. For more information, see the [JVM Dump API documentation](https://www.ibm.com/support/knowledgecenter/SSYKE2_8.0.0/com.ibm.java.api.80.doc/com.ibm.jvm.ras/index.html?view=kc).
+
+## Analyzing dumps
+
+The best method to analyze a PHD heap dump is to use the [Eclipse Memory Analyzer tool (MAT)](https://www.eclipse.org/mat/) or the [IBM Memory Analyzer tool](https://www.ibm.com/support/knowledgecenter/en/SSYKE2_8.0.0/com.ibm.java.80.doc/diag/tools/tool_memoryanalyzer.html). These tools process the dump file and provide a visual representation of the objects in the Java Heap. Both tools require the Diagnostic Tool Framework for Java (DTFJ) plugin. To install the DTFJ plugin in the Eclipse IDE, select the following menu items:
 
 ```
 Help > Install New Software > Work with "IBM Diagnostic Tool Framework for Java" > IBM Monitoring and Diagnostic Tools > Diagnostic Tool Framework for Java   
 ```
+The following sections contain detailed information about the content of each type of heap dump file.
+
+### Portable Heap Dump (PHD) format
+
+A PHD format dump file contains a header section and a body section. The body section can contain information about object, array, or class records.
+Primitive numbers are used to describe the file format, as detailed in the following table:
+
+| Primitive number | Length in bytes                              |
+|------------------|----------------------------------------------|
+| `byte`           | 1                                            |
+| `short`          | 2                                            |
+| `int`            | 4                                            |
+| `long`           | 8                                            |
+| `word`           | 4 (32-bit platforms) or 8 (64-bit platforms) |
+
+#### General structure
+
+The following structure comprises the header section of a PHD file:
+
+- A UTF string indicating that the file is a `portable heap dump`
+- An `int` containing the PHD version number
+- An `int` containing flags:
+    - `1` indicates that the `word` length is 64-bit.
+    - `2` indicates that all the objects in the dump are hashed. This flag is set for heap dumps that use 16-bit hash codes. These heap dumps use 32-bit hash codes that are only created when used. For example, these hash codes are created when the APIs `Object.hashCode()` or `Object.toString()` are called in a Java application. If this flag is not set, the presence of a hash code is indicated by the hash code flag on the individual PHD records.
```

Comment:
This is a bit confusing. I believe it's saying that flag 2 is obsolete, but this isn't explicitly mentioned.

Reply:
Not just me then. I was going to ask you what it meant! Should I just replace the words with "obsolete"?

Reply:
I don't like the `These` in `These heap dumps use ...`. I'd replace it with `Current` or something similar.

Reply:
Or `OpenJ9`

Reply:
Updated as suggested.